Fix failing Collection CI Checks

djyasin opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Fix the following failing collection CI checks:
  - CI / awx_collection integration (a-h, ^[a-h])
  - CI / awx_collection integration (i-p, ^[i-p])
  - CI / awx_collection integration (r-z0-9, ^[r-z0-9])

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does. Also please make sure that if this PR has an attached JIRA, put AAP-<number>
in as the first entry for your PR title.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - Other



##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```
awx: 4.6.20
```


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Standardizes CI on Python 3.13 and hardens collection integration/coverage by fixing Python/ansible-test resolution and environment setup.
> 
> - **CI/Workflows**:
>   - Standardize Python to `3.13` across `dev-env`, `awx-operator`, and `collection-integration` jobs.
>   - Replace `python3 -m pip` with `python -m pip`; add `hash -r` and adjust `PATH` to ensure `ansible-test` resolution.
>   - Uninstall system `ansible`/`ansible-core` before installs to avoid conflicts; explicitly upgrade `ansible-core`.
>   - For collection integration: export `PYTHONPATH` to include site-packages; upload and combine coverage with explicit Python bin `PATH`.
> - **GitHub Actions**:
>   - `run_awx_devel`: use `python -m pip` for upgrading `ansible-core`.
>   - `awx_devel_image`: remove redundant `setup-python` step.
> - **Makefile**:
>   - Add `ANSIBLE_TEST_PYTHON_VERSION` (default `3.13`) and pass `--python` to `ansible-test integration`.
>   - Prefix `ansible-test` invocations with Python bin in `PATH` for integration coverage.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit d914625a1514e78874538b7e80a0a455aff17a72. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

djyasin committed:
Dummy change to see if awx-operator ci check is broken in devel

AlanCoding commented:
Documenting outcome:

> This version of ansible-test cannot be executed with Python version 3.14.0. Supported Python versions are: 3.11, 3.12, 3.13

djyasin committed:
Fix CI: Use Python 3.13 for ansible-test compatibility

  ansible-test only supports Python 3.11, 3.12, and 3.13.
  Changed collection-integration jobs from '3.x' to '3.13'
  to avoid using Python 3.14 which is not supported.

djyasin committed:
Revert dummy change in README

AlanCoding reviewed (approved):
This probably works. I am aware python 3.14 recently became official and that could easily explain what we saw. It if fixes it, you can merge.

djyasin committed:
Fix ansible-test Python version for CI integration tests

  ansible-test only supports Python 3.11, 3.12, and 3.13.
  Added ANSIBLE_TEST_PYTHON_VERSION variable to explicitly pass
  --python 3.13 flag to ansible-test integration command.

  This prevents ansible-test from auto-detecting and using
  Python 3.14.0, which is not supported.

djyasin committed:
Fix CI: Execute ansible-test with Python 3.13 to avoid
  unsupported Python 3.14

djyasin committed:
Fix CI: Use Python 3.13 across all jobs to avoid Python
  3.14 compatibility issues

djyasin committed:
Fix CI: Use 'python' and 'ansible.test' module for Python
   3.13 compatibility

AlanCoding commented:
now seeing

```
Installing 'awx.awx:100.100.100-git' to '/home/runner/.ansible/collections/ansible_collections/awx/awx'
awx.awx:100.100.100-git was installed successfully
cd /home/runner/.ansible/collections/ansible_collections/awx/awx && \
	python -m ansible.test integration --python 3.13 --coverage -vvv --requirements ad_hoc_command ad_hoc_command_cancel ad_hoc_command_wait bulk_host_create bulk_host_delete bulk_job_launch credential credential_input_source credential_type demo_data execution_environment export group host  && \
	python -m ansible.test coverage xml --requirements --group-by command --group-by version
/opt/hostedtoolcache/Python/3.14.0/x64/bin/python: No module named ansible.test
make: *** [Makefile:435: test_collection_integration] Error 1
```

still looks like a python 3.14 problem

djyasin committed:
Fix CI: Use 'python' instead of 'python3' for Python 3.13
   compatibility

AlanCoding commented:
Looking better! Here's a good hint from the recent output:

> Error while resolving value for '_rrule': The lookup plugin 'awx.awx.schedule_rrule' failed to load: No module named 'pytz'

I'm concerned there's still something funky going on with the default Github Actions. IIRC, it is Ubuntu, and has ansible installed by default. So my concern is hitting the default install, perhaps `ansible-test` as it exists in the runner before any of our stuff has happened. Maybe this could be protected against by uninstalling `ansible` at the very start of the action, to prevent the confusion.

AlanCoding commented:
I would try `python -m ansible.test` instead of the direct `ansible-test`. I trust the python `-m` more because it assures we're in the python environment, and I'm almost sure there's another evil version of Ansible somewhere else on the machine.

djyasin committed:
Fix CI: Ensure ansible-test uses Python 3.13 environment
  explicitly

djyasin committed:
Fix: Remove silent failure check for ansible-core in test suite

djyasin committed:
Fix CI: Export PYTHONPATH to make awxkit available to ansible-test

djyasin committed:
Fix CI: Use 'python' in run_awx_devel to maintain Python
  3.13 environment

djyasin committed:
Fix CI: Remove setup-python from awx_devel_image that was resetting Python 3.13 to 3.14

tvo318 reviewed (approved):
LGTM. Thanks for doing this!
